src/main/java/kr/hhplus/be/server/infrastructure/config/redis/DistributedLockService.java

```diff
@@ -0,0 +1,109 @@
+package kr.hhplus.be.server.infrastructure.config.redis;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.redisson.api.RLock;
+import org.redisson.api.RedissonClient;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class DistributedLockService {
+
+    private final RedissonClient redissonClient;
+
+    /**
+     * 분산락을 사용하여 작업을 실행
+     * @param lockKey 락 키
+     * @param waitTime 락 대기 시간 (초)
+     * @param leaseTime 락 유지 시간 (초)
+     * @param task 실행할 작업
+     * @return 작업 결과
+     */
+    public <T> T executeWithLock(String lockKey, long waitTime, long leaseTime, Supplier<T> task) {
+        RLock lock = redissonClient.getLock(lockKey);
+        
+        try {
+            log.debug("락 획득 시도 - 키: {}", lockKey);
+            
+            if (lock.tryLock(waitTime, leaseTime, TimeUnit.SECONDS)) {
+                try {
+                    log.debug("락 획득 성공 - 키: {}", lockKey);
+                    return task.get();
+                } finally {
+                    lock.unlock();
```

Comment:
```suggestion
                    if (lock.isHeldByCurrentThread())
```

현재 스레드가 락을 가지고 있는지도 체크해보면 좋겠네요!